From 2b183bdd51f353eba4379e2aa2b190680362b274 Mon Sep 17 00:00:00 2001
From: Pascal Quantin <pascal.quantin@gmail.com>
Date: Wed, 24 Jun 2015 18:04:13 -0700
Subject: [PATCH 04/12] Fix proto_tree_add_bytes_item() when we fail to decode
 number from string

We still need to insert an empty g_byte_array

Bug: 11309
Change-Id: Idf2ed8ea81b3f612ef8d6ae1aed158108dcc4356
Reviewed-on: https://code.wireshark.org/review/9114
Reviewed-by: Pascal Quantin <pascal.quantin@gmail.com>
(cherry picked from commit 6126a6455058696dd0ac2073032bdfe066a6ae38)
Reviewed-on: https://code.wireshark.org/review/9115
(cherry picked from commit 2728dfcf6f388f2a1283d9f22bce5bc66a24d22f)
Reviewed-on: https://code.wireshark.org/review/10582
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
---
 epan/proto.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/epan/proto.c b/epan/proto.c
index 1169025..a1a34a4 100644
--- a/epan/proto.c
+++ b/epan/proto.c
@@ -2108,6 +2108,8 @@ proto_tree_add_bytes_item(proto_tree *tree, int hfindex, tvbuff_t *tvb,
 
 		if (bytes)
 		    proto_tree_set_bytes_gbytearray(new_fi, bytes);
+		else
+		    proto_tree_set_bytes(new_fi, NULL, 0);
 
 		if (created_bytes)
 		    g_byte_array_free(created_bytes, TRUE);
-- 
2.1.4

